rdkit/Chem/Pharm2D/UnitTestMatcher.py

```diff
@@ -28,10 +31,7 @@ def setUp(self):
     self.factory.SetBins([(0, 2), (2, 5), (5, 8)])
     self.factory.Init()
 
-  def test1(self):
-    """ simple tests
-
-    """
+  def test1_simple(self):
```

Comment:
I use verbose output for the tests. This looks normally like this. Last four lines are for running the doctests. 
```
test1MurckoScaffold (Scaffolds.UnitTestMurckoScaffold.TestCase) ... ok
test2MurckoScaffold (Scaffolds.UnitTestMurckoScaffold.TestCase) ... ok
test_MakeScaffoldGeneric (Scaffolds.UnitTestMurckoScaffold.TestCase) ... ok
test_MurckScaffoldSmilesFromSmiles (Scaffolds.UnitTestMurckoScaffold.TestCase) ... ok
test_MurckoScaffoldSmiles (Scaffolds.UnitTestMurckoScaffold.TestCase) ... ok
test_ReferenceImplementation (Scaffolds.UnitTestMurckoScaffold.TestCase) ... ok
GetScaffoldForMol (rdkit.Chem.Scaffolds.MurckoScaffold)
Doctest: rdkit.Chem.Scaffolds.MurckoScaffold.GetScaffoldForMol ... ok
MakeScaffoldGeneric (rdkit.Chem.Scaffolds.MurckoScaffold)
Doctest: rdkit.Chem.Scaffolds.MurckoScaffold.MakeScaffoldGeneric ... ok
```
If there is a docstring, that information is also printed:
```
test1LazyReader (Suppliers.UnitTestSmilesMolSupplier.TestCase)
tests lazy reads ... ok
test2LazyIter (Suppliers.UnitTestSmilesMolSupplier.TestCase)
tests lazy reads using the iterator interface ... ok
```
As the docstring is often just a repetition of the method name and I therefore prefer to remove the redundancy. 

I just tried adding the '-v' flag to the TestRunner for Python codes, however that does only print `__main__.TestCase` every time as each test is run individually. 
```
116: test1 (__main__.TestCase) ... ok
116: test10 (__main__.TestCase) ... ok
116: test11 (__main__.TestCase) ... ok
116: test12 (__main__.TestCase) ... ok
116: test13 (__main__.TestCase) ... ok
```
My proposal is to run all the Python tests in `rdkit.Chem` using automatic test detection. Essentially:
```
unittest.TestLoader().discover('rdkit/Chem', pattern='UnitTest*')
```
No need to keep maintaining the test_all scripts and no way of loosing tests over time. Added advantage is, it's easy to run coverage analysis. I actually already do this and for this branch have in rdkit.Chem
```
----------------------------------------------------------------------
Ran 325 tests in 31.868s

FAILED (errors=4, skipped=1)
```
Using the TestRunner, I get:
```
1/1 Test #116: pythonTestDirChem ................   Passed   56.23 sec
```
I still miss a few doctests mainly in the rdkit.Chem directory (e.g. BRICS.py). It would be interesting to see how much restarting of Python processes costs.

For all the rdkit subdirectories, I find currently:
```
Ran 608 tests in 34.697s

FAILED (errors=4, skipped=13)
```